internal/pkg/earpointer/analysis.go

```diff
@@ -315,7 +350,139 @@ func (vis *visitor) visitSelect(s *ssa.Select) {
+	for arg, params := range paramCstrs {
+		for _, param := range params {
+			// The copy-by-value semantics is implemented by the SSA call.
+			// For example:
+			//   type S struct { ... }
+			//   var v S
+			//   g(v)
+			//   func g(a S) { ... }
+			// SSA creates a copy of v and passes it to g. Hence we directly
+			// unify this copy and g's argument "a".
+			if mayShareObject(arg) {
+				vis.unifyLocals(arg, param)
+			}
+		}
+	}
+	// Handle return value unification.
+	for callerRet, rets := range retCstrs {
+		for _, calleeRet := range rets {
+			if len(calleeRet) == 1 { // non-tuple case
+				if mayShareObject(calleeRet[0]) {
+					vis.unifyLocals(callerRet, calleeRet[0])
+				}
+			} else { // tuple case
+				// For call "return_r = f(...)", if the callee returns a tuple
+				// <r0, r1, ...>, then the i^{th} element in the tuple is
+				// connected to the caller's return_r by making
+				// "return_r[i] = r{i}".
+				for k := 0; k < len(calleeRet); k++ {
+					retV := calleeRet[k]
+					if !mayShareObject(retV) {
+						continue
+					}
+					// Implement "return_r[i] = r{i}".
+					field := Field{Name: strconv.Itoa(k)}
+					vis.processHeapAccess(retV, callerRet, field)
+				}
+			}
+		}
+	}
+}
+
+// Handle calls to builtin functions: https://golang.org/pkg/builtin/.
+func (vis *visitor) visitBuiltin(builtin *ssa.Builtin, instr ssa.Instruction) {
+	switch builtin.Name() {
+	case "append": // func append(slice []Type, elems ...Type) []Type
+		// Propagage the arguments to the return value.
+		if dst, ok := instr.(ssa.Value); ok {
+			ops := instr.Operands(nil)
+			// ops[0] is the "append" function itself.
+			for i := 1; i < len(ops); i++ {
+				// Unify-by-reference the dst and the op for all contexts.
+				vis.unifyLocals(dst, *ops[i])
+			}
+		}
+	case "copy": // "func copy(dst, src []Type) int"
+		{
+			// Propagage the source to the destination through unify-by-value.
+			ops := instr.Operands(nil)
+			// ops[0] is the "copy" function itself.
+			dst, src := *ops[1], *ops[2]
+			if mayShareObject(src) {
+				for _, c := range vis.contexts {
+					vis.unifyByValue(MakeLocal(c, dst), MakeLocal(c, src))
+				}
+			}
+		}
+	}
+}
+
+// Collect unification constraints corresponding to a call.
+// This generates constraints for unifying parameters, free variables, and return values.
+func (vis *visitor) collectCalleeConstraints(common *ssa.CallCommon, fn *ssa.Function, instr ssa.Instruction) (map[ssa.Value][]ssa.Value, map[ssa.Value][][]ssa.Value) {
+	// Handle undefined/unknown functions.
+	if fn == nil || len(fn.Blocks) == 0 || len(fn.Params) != len(common.Args) {
+		return nil, nil
+	}
+
+	paramCstrs := make(map[ssa.Value][]ssa.Value)
+	// Add caller_reg -> {callee_reg} constraints for parameters.
+	// For example, for g(a, b) and func g(x, y), add <a, g.x> and
+	// <b, g.y> into the constraints.
+	// Disable the unification of receivers here to avoid too many FPs.
```

Comment:
Why would this lead to too many FPs?

Reply:
Add comments. This will be addressed in the follow-up PR(s).